### ISSUE TITLE
Keyspace not injected in all UDT meta properties

### ISSUE THREAD

**Issue by goyalzz** [doanduyhai/Achilles#276]
Hi @doanduyhai,
I got an error of default keyspace error for UDT Creation while creating List<frozen<UDT>> in Achillies version 5.1.4.

Below Code for UDT Creation:
```
@UDT
public class Goal {

	@Column
	private Long goalId;

	@Column
	private String goalName;

	public Long getGoalId() {
		return goalId;
	}

	public void setGoalId(Long goalId) {
		this.goalId = goalId;
	}

	public String getGoalName() {
		return goalName;
	}

	public void setGoalName(String goalName) {
		this.goalName = goalName;
	}
}
```

```List<@Frozen Goal> goals;```

Got Below Exception:
```
11:28:18.945 [main] DEBUG org.springframework.beans.factory.support.DisposableBeanAdapter - Invoking destroy() on bean with name 'org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor'
Exception in thread "main" org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jerseyResourceConfig': Invocation of init method failed; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserController': Unsatisfied dependency expressed through field 'advertiserService': Error creating bean with name 'advertiserServiceImpl': Unsatisfied dependency expressed through field 'advertiserRepository': Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserServiceImpl': Unsatisfied dependency expressed through field 'advertiserRepository': Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:136)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:408)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1570)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:545)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:776)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:861)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:541)
	at com.prontoitlabs.mobiadz.Main.startServer(Main.java:45)
	at com.prontoitlabs.mobiadz.Main.main(Main.java:35)
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserController': Unsatisfied dependency expressed through field 'advertiserService': Error creating bean with name 'advertiserServiceImpl': Unsatisfied dependency expressed through field 'advertiserRepository': Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserServiceImpl': Unsatisfied dependency expressed through field 'advertiserRepository': Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:569)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:349)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1214)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:543)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeansWithAnnotation(DefaultListableBeanFactory.java:581)
	at org.springframework.context.support.AbstractApplicationContext.getBeansWithAnnotation(AbstractApplicationContext.java:1206)
	at com.prontoitlabs.mobiadz.config.JerseyResourceConfig.setup(JerseyResourceConfig.java:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:365)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:310)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:133)
	... 13 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserServiceImpl': Unsatisfied dependency expressed through field 'advertiserRepository': Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:569)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:349)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1214)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:543)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:207)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1214)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1054)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1019)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:566)
	... 32 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'advertiserRepository': Unsatisfied dependency expressed through field 'managerFactory': Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:569)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:349)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1214)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:543)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:207)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1214)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1054)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1019)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:566)
	... 46 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraNativeClusterProduction' defined in com.prontoitlabs.mobiadz.config.CassandraConfig: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1123)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1018)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:510)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:207)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1214)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1054)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1019)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:566)
	... 60 more
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [info.archinnov.achilles.generated.ManagerFactory]: Factory method 'cassandraNativeClusterProduction' threw exception; nested exception is info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588)
	... 73 more
Caused by: info.archinnov.achilles.exception.AchillesException: The keyspace name for the UDT type 'com.prontoitlabs.mobiadz.domain.Apps' should be either provided by the 'UDT' annotation or at runtime
	at info.archinnov.achilles.validation.Validator.validateTrue(Validator.java:119)
	at info.archinnov.achilles.internals.metamodel.AbstractUDTClassProperty.buildType(AbstractUDTClassProperty.java:114)
	at info.archinnov.achilles.internals.metamodel.AbstractUDTClassProperty.inject(AbstractUDTClassProperty.java:157)
	at info.archinnov.achilles.internals.metamodel.UDTProperty.inject(UDTProperty.java:153)
	at info.archinnov.achilles.internals.metamodel.ListProperty.inject(ListProperty.java:199)
	at info.archinnov.achilles.internals.metamodel.AbstractEntityProperty.inject(AbstractEntityProperty.java:457)
	at info.archinnov.achilles.internals.context.ConfigurationContext.injectDependencies(ConfigurationContext.java:337)
	at info.archinnov.achilles.internals.runtime.AbstractManagerFactory.lambda$injectDependencies$464(AbstractManagerFactory.java:178)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at info.archinnov.achilles.internals.runtime.AbstractManagerFactory.injectDependencies(AbstractManagerFactory.java:178)
	at info.archinnov.achilles.internals.runtime.AbstractManagerFactory.bootstrap(AbstractManagerFactory.java:119)
	at info.archinnov.achilles.generated.ManagerFactory.<init>(ManagerFactory.java:133)
	at info.archinnov.achilles.generated.ManagerFactoryBuilder.build(ManagerFactoryBuilder.java:38)
	at com.prontoitlabs.mobiadz.config.CassandraConfig.cassandraNativeClusterProduction(CassandraConfig.java:65)
	at com.prontoitlabs.mobiadz.config.CassandraConfig$$EnhancerBySpringCGLIB$$a7732153.CGLIB$cassandraNativeClusterProduction$0(<generated>)
	at com.prontoitlabs.mobiadz.config.CassandraConfig$$EnhancerBySpringCGLIB$$a7732153$$FastClassBySpringCGLIB$$a16153f6.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356)
	at com.prontoitlabs.mobiadz.config.CassandraConfig$$EnhancerBySpringCGLIB$$a7732153.cassandraNativeClusterProduction(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 74 more
```

Please help me to solve this Exception, Thanks.

**Comment by doanduyhai** [comment]
`The keyspace name for the UDT type 'com.domain.Apps' should be either provided by the 'UDT' annotation or at runtime`

The error message is very clear, since you did not specify default keyspace on UDT annotation, you have to define the default keyspace when bootstrapping **Achilles**:


```java
     ManagerFactoryBuilder
       .builder(cluster)
       .withDefaultKeyspaceName("xxx")
       ...

```

**Comment by goyalzz** [comment]
@doanduyhai,
I already pass default keyspace in below code:
```
ManagerFactory factory = ManagerFactoryBuilder
				.builder(cluster)
				.withDefaultKeyspaceName(
						"mobiadz")
				.doForceSchemaCreation(true).build();
```

**Comment by goyalzz** [comment]
Hi @doanduyhai,

Can you please help me to resolve this problem?

**Comment by doanduyhai** [comment]
Are you using `SchemaNameProvider` ? 

Also can you please paste here source code of DSL using schemaNameProvider that causes the exception ?

**Comment by goyalzz** [comment]
@doanduyhai ,
This Error occur while creating tables by using below code:
```
ManagerFactory factory = ManagerFactoryBuilder
                .builder(cluster)
                .withDefaultKeyspaceName(
                        "mobiadz")
                .doForceSchemaCreation(true).build();
```
It could not able to create table having ```List<@Frozen UDT>``` if the UDT have not mention any keyspace name in @UDT annotation.

**Comment by doanduyhai** [comment]
Thanks @goyalzz I'll try to reproduce your issue and attempt a fix/release

**Comment by goyalzz** [comment]
Hi @doanduyhai ,
Please help me how may i fix this issue.

**Comment by doanduyhai** [comment]
Issue successfully reproduced

**Comment by doanduyhai** [comment]
Bug found, easy fix, expect a release today

**Comment by doanduyhai** [comment]
Fixed in Achilles 5.1.5

**Comment by goyalzz** [comment]
Thanks Alot.
